sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ConfigurationSamples.cs

```diff
@@ -0,0 +1,189 @@
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Update a twin configuration.
+        /// </summary>
+        /// <param name="twinConfiguration">Twin Configuration to to be updated.</param>
+        public async Task<TwinConfiguration> UpdateConfiguratioAsync(TwinConfiguration twinConfiguration)
+        {
+            SampleLogger.PrintHeader("UPDATE A CONFIGURATION");
+
+            try
+            {
+                #region Snippet:IotHubUpdateConfiguration
+
+                twinConfiguration.Priority = Random.Next(MaxRandomValue);
+                Console.WriteLine($"Updating twin configuration with Id: '{twinConfiguration.Id}''s priority to: '{twinConfiguration.Priority}'");
+                Response < TwinConfiguration > response = await IoTHubServiceClient.Configurations.CreateOrUpdateConfigurationAsync(twinConfiguration, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+
+                TwinConfiguration updatedConfiguration = response.Value;
+
+                SampleLogger.PrintSuccess($"Successfully updated twin configuration: Id: '{updatedConfiguration.Id}', Priority: '{updatedConfiguration.Priority}', ETag: '{updatedConfiguration.Etag}'");
+
+                #endregion Snippet:IotHubUpdateConfiguration
+
+                return updatedConfiguration;
+            }
+            catch (Exception ex)
+            {
+                // Try to cleanup before exiting with fatal error.
+                await DeleteConfiguratioAsync(twinConfiguration);
+                SampleLogger.FatalError($"Failed to update a twin configuration due to:\n{ex}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Deletes a twin configuration.
+        /// </summary>
+        /// <param name="twinConfiguration">Twin Configuration to deletec.</param>
+        public async Task DeleteConfiguratioAsync(TwinConfiguration twinConfiguration)
+        {
+            SampleLogger.PrintHeader("DELETE A CONFIGURATION");
+
+            try
+            {
+
+                Console.WriteLine($"Deleting twin configuration with Id: '{twinConfiguration.Id}'");
+
+                #region Snippet:IotHubDeleteConfiguration
+
+                Response response = await IoTHubServiceClient.Configurations.DeleteConfigurationAsync(twinConfiguration, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+
+                SampleLogger.PrintSuccess($"Successfully deleted twin configuration with Id: '{twinConfiguration.Id}'");
+
+                #endregion Snippet:IotHubDeleteConfiguration
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to delete twin configuration due to:\n{ex}");
+            }
+        }
+
+        private TwinConfiguration CreateSampleConfig(string testConfigurationId)
+        {
+            var twinConfiguration =
```

Comment:
Preferred code layout:

```
var twinConfiguration = new TwinConfiguration
{
    Id = testConfigurationId,
};
```

In some cases when there is just one property, we also do:

```
var twinConfiguration = new TwinConfiguration { Id = testConfigurationId };
```